documentation/authentication.md

```diff
@@ -0,0 +1,135 @@
+
+## API Token Authentication
+An API Token is the recommended authentication mechanism for `dotnet-monitor`. To enable it, you will need to generate a secret key, update the configuration of `dotnet monitor`, and then specify the API Token in the Authorization header on all requests to `dotnet-monitor`.
+
+> **NOTE:** API Token Authentication should only be used when [TLS is enabled](#) to protect the token while in transit.
+
+### Generating an API Token
+
+The API Token you use to secure `dotnet-monitor` should be a 32-bit cryptographically random secret. You can generate a token either using `dotnet-monitor` or via your shell. To generate an API Token with `dotnet-monitor`, simply invoke the `generatekey` command:
+
+```powershell
+dotnet-monitor generatekey
+```
+
+The output from this command will display the API Token formatted as an authentication header along with its hash and associated hashing algorithm. You will need to store the `ApiKeyHash` and `ApiKeyHashType` in the configuration for `dotnet-monitor` and use the authorication header value when making requests to the `dotnet-monitor` HTTPS endpoint.
+
+```
+Authorization: MonitorApiKey Uf6Yq8ZGkcn+ltq2QLHxuXpKA6/Q5VH5Mb5aSIJBxRc=
+ApiKeyHash: CB233C3BE9F650146CFCA81D7AA608E3A3865D7313016DFA02DAF82A2505C683
+ApiKeyHashType: SHA256
+```
+
+> NOTE: The generated API token should be secured at rest. We recommend using a tool such as a password manager to save it.
+
+### Configuring dotnet-monitor to use an API Token
+
+Using the `generatekey` command does not automatically update the configuration of `dotnet-monitor` to use the new key. You can update the configuration of `dotnet-monitor` via settings file, environemt variable of kubernetes secrets.
+
+If you're running on Windows, you can save these settings to `%USERPROFILE%\.dotnet-monitor\settings.json`. If you're on other operating systems, you can save this file to `$XDG_CONFIG_HOME/dotnet-monitor/settings.json`.
+
+> **NOTE:** If `$XDG_CONFIG_HOME` isn't defined, dotnet-monitor will fall back to looking at `$HOME/.config/dotnet-monitor/settings.json`
+
+```json
+{
+  "ApiAuthentication": {
+    "ApiKeyHash": "<HASHED-TOKEN>",
+    "ApiKeyHashType": "SHA256"
+  }
+}
+```
+
+Alternatively, you can use environment variables to specify the configuration.
+
+```sh
+DotnetMonitor_ApiAuthentication__ApiKeyHash="<HASHED-TOKEN>"
+DotnetMonitor_ApiAuthentication__ApiKeyHashType="SHA256"
+```
+
+> **NOTE:** When you use environment variables to configure the API Key hash, you must restart `dotnet-monitor` for the changes to take effect.
+
+### Configuring an API Token in a Kubernetes Cluster
+If you're running in Kubernetes, we recommend creating secrets and mounting them into container via a deployment manifest.
+
+```sh
+kubectl create secret generic apikey \
```

Comment:
note this command can also be used to do a key rotation. Not sure you want to mention that only secrets mounted as volumes will support key rotations without restarts.